File: app/src/main/java/com/test/kabak/openweather/core/network/SearchCityResponse.java
package com.test.kabak.openweather.core.network;

import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;

public class SearchCityResponse {
    @SerializedName("_embedded")
    public Embedded embedded;

    public static class Embedded {
        @SerializedName("city:search-results")
        public ArrayList<SearchResult> searchList;
    }

    public static class SearchResult {
        @SerializedName("matching_full_name")
        public String name;
        @SerializedName("_links")
        public Links links;
    }

    public static class Links {
        @SerializedName("city:item")
        public CityItem cityItem;
    }

    public static class CityItem {
        @SerializedName("href")
        public String link;
    }
}

File: app/src/main/java/com/test/kabak/openweather/core/network/dataClasses/SearchCityResponse.kt
package com.test.kabak.openweather.core.network.dataClasses

import com.google.gson.annotations.SerializedName

class SearchCityResponse(
    @SerializedName("_embedded") val embedded: Embedded,
) {
    class Embedded(
        @SerializedName("city:search-results")
        val searchList: ArrayList<SearchResult>,
    )

    class SearchResult(
        @SerializedName("matching_full_name") val name: String,
        @SerializedName("_links") val links: Links,
    )

    class Links(
        @SerializedName("city:item") val cityItem: CityItem,
    )

    data class CityItem(
        @SerializedName("href") val link: String,
    )
}